lib/asn1/gen.c

```diff
@@ -1694,13 +1711,15 @@ declare_type(const Symbol *s, Type *t, int typedefp)
 {
     char *newbasename = NULL;
 
-    if (typedefp)
+    if (typedefp && !s->emitted_declaration)
```

Comment:
This change is not necessary.  I'll remove it.

Reply:
I tested without this change and I concur.